tests/Insightly/Listeners/SyncContactTest.php

```diff
@@ -0,0 +1,332 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Insightly\Listeners;
+
+use App\Domain\Contacts\Contact;
+use App\Domain\Contacts\ContactType;
+use App\Domain\Contacts\Events\ContactCreated;
+use App\Domain\Contacts\Events\ContactUpdated;
+use App\Domain\Contacts\Repositories\ContactRepository;
+use App\Insightly\InsightlyMapping;
+use App\Insightly\Listeners\SyncContact;
+use App\Insightly\Repositories\InsightlyMappingRepository;
+use App\Insightly\Resources\ResourceType;
+use Iterator;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\NullLogger;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+use Tests\MockInsightlyClient;
+
+final class SyncContactTest extends TestCase
+{
+    use MockInsightlyClient;
+
+    private SyncContact $syncContact;
+    private UuidInterface $integrationId;
+    private int $insightlyIntegrationId;
+    private UuidInterface $contactId;
+    private int $insightlyContactId;
+    private string $contactEmail;
+    private ContactRepository&MockObject $contactRepository;
+    private InsightlyMappingRepository&MockObject $insightlyMappingRepository;
+
+    protected function setUp(): void
+    {
+        $this->integrationId = Uuid::uuid4();
+        $this->insightlyIntegrationId = 111;
+
+        $this->contactId = Uuid::uuid4();
+        $this->insightlyContactId = 222;
+        $this->contactEmail = 'info@publiq.be';
+
+        $this->mockCrmClient();
+        $this->contactRepository = $this->createMock(ContactRepository::class);
+        $this->insightlyMappingRepository = $this->createMock(InsightlyMappingRepository::class);
+
+        $this->syncContact = new SyncContact(
+            $this->insightlyClient,
+            $this->contactRepository,
+            $this->insightlyMappingRepository,
+            new NullLogger(),
+        );
+    }
+
+    public function test_it_does_not_create_a_contributor(): void
+    {
+        $this->givenThereIsAContactForAnIntegration(ContactType::Contributor);
+
+        $this->insightlyClient->expects($this->never())
+            ->method('contacts');
+
+        $this->syncContact->handleContactCreated(new ContactCreated($this->contactId));
+    }
+
+    public function test_it_does_not_update_a_contributor(): void
+    {
+        $this->givenThereIsAContactForAnIntegration(ContactType::Contributor);
+
+        $this->insightlyClient->expects($this->never())
+            ->method('contacts');
+
+        $this->syncContact->handleContactUpdated(new ContactUpdated($this->contactId, true));
+    }
+
+    public function test_it_links_a_new_insightly_contact_when_platform_contact_was_created(): void
```

Comment:
What is `platform` here?

Reply:
`publiq platform`. Should I name it that explicit?

Reply:
Or just remove it?